### ISSUE TITLE
Maybe paper isn't the endpoint

### ISSUE THREAD

**Issue by cranmer** [betatim/openscienceprize#50]
Hi,

Just read the new draft proposal. Looks good. I like the beginning for sure.

One thing that strikes me is that I think it is a trap of the current scientific model to think of papers as the endpoint. That's part of the mind-set that lead to our reproducibility and openness problem in the first place. And it also makes composition harder.

I'd suggest that we think of the result of one of these reusable research products as workflow that can be extended or embedded into a larger workflow. The work that Lukas and I have been doing is based on the idea of a "workflow template" or "parametrized workflow", which has been quite a flexible model. As such, the paper might just be one of the outputs products ("entities" in the PROV language), but along with that would be other products.  There could be multiple papers, multiple data products, etc.

I think this is all in agreement with the what others are thinking, but proposal has a lot of focus on papers:
" with tools to go from an empty directory to a fully rendered paper "
"Despite these advances, there are still many missing components of a system for executable papers."
"paper repository"

**Comment by betatim** [comment]
Agree we should get away from the PDF on arxiv being the (only) research output. In my mind the object to share is the repository that contains "all the things" (including info on access to the data, what kind of computing you need, etc). It should contain a machine readable description of what the repository produces and how to use what is in the repository to produce those artefacts.

What people do inside the repository should be up to them. Do they want to use python or R or c++. `make` or a shell script that strings things together, or ... The main motivation for this is that there are so many different styles of research project, with many different tools in existence, and opinions on how you should do it. In addition people get attached to their tools.

Why is a "paper" special? I think something that is a summary of what went on, for human consumption, serves as an advert for the work, and gives you an entry point to understanding how the main results came about is a necessary but not sufficient output.

It is perfectly acceptable to say in your executable paper "to reproduce table 5, execute this part of our workflow graph" (what ever the command for that might be). The key is that straight after the command you display table 5 (or you state in the caption how to obtain the table)

**Comment by cranmer** [comment]
I'm not against papers, but it's not like we need to advocate for papers, it's already the dominant form of research output. 

But Table 5 might be on more than one paper. And using the paper as a conduit for a generic workflow has lots of problems. 

So I would advocate that we think of these everpub products as reusable/executable workflows, for which the paper is one possible form of output.

**Comment by rougier** [comment]
In such reusable/executable workflow, paper could be considered as a specific "release" worth being officially published because they carry important results/analysis/facts. Whatever the form, we need to have these "singular" points in the workflow.

**Comment by betatim** [comment]
Even if you publish (as in make public) a workflow (which forms part of a
larger thing that then gets published in a traditional journal article) you
need to attach a README to it, this should be something better than a text
document or PDF. I'd already be happy if every science result/factoid that
is being made public had an executable README.

Not sure what we should call it, as `paper` is a term that has a very
specific meaning to academics already.

**Comment by anaderi** [comment]
I agree with the point that papers are just a snapshots of something bigger (research flow?), and probably during research, making such snapshots should be kind of routine task, but would it make sense to stress it in this proposal without risking to drag reader's attention from the main focus? If it is possible to mention this dynamics: research flow -> workflow pipeline -> papers as snapshots, I'd go for it. The proposal looks to me rather consistent now and I'm not sure my English would allow improving it.

**Comment by cranmer** [comment]
I will try to edit the proposal to reflect this discussion with papers being one type of resulting product ("entity") in the [PROV language](https://www.w3.org/TR/prov-overview/) if there is general agreement. Otherwise, I won't bother.

**Comment by ctb** [comment]
+1 from me!

**Comment by khinsen** [comment]
+1 from me as well. But I am not sure this is an important point at the current stage, i.e. the proposal we are submitting. It's more about the overall evolution of the scientific publication system than about our concrete contribution.

**Comment by Repositorian** [comment]
Couple notes from library land on object model ( paper v. event v. workflow): the endpoint when it comes to the social aspects of knowledge production is fixity. Citation, attribution, contributorship, Intellectual property all rely on something fixed to attach to. The RDA is looking at this issue thru the lens of citing dynamic datasets but the solution looks like a call to make something at least ephemerally  static on the fly. At the end of the day, there is something essential with fixity that humans and machines need to reference in order to attach claims, metadata, licenses.

**Comment by cranmer** [comment]
Nice points!

the OSF organized by project, which is composed of many different
components. You can register individual components and get a DOI, or you
can release the entire project. You can fork projects, embed them in other
projects (composition), etc.

I think that we can think about the entire everpub project (its workflow
and outputs) as one entity that can be fixed for a DOI while letting people
compose into larger projects or fork for modification and/or remixing of
individual components.

In short, fix the project and don't focus on one specific output of the
workflow (eg the paper)

Kyle

**Comment by khinsen** [comment]
+1

That is exactly the point of view I adopted for [ActivePapers](http://www.activepapers.org/). The DOI-referenced entity is a combination of data, code, and narrative, with all parts optional of course. _But_ there is also a need to refer to items inside such a bundle in a standard way, not so much for citing but for reuse. That's something we have to define as part of the data model for the paper/project/whatever we call it.
